src/box/memtx_tree.cc

```diff
@@ -1350,6 +1358,8 @@ memtx_tree_index_create_iterator(struct index *base, enum iterator_type type,
 		it->key_data.set_hint(key_hint(key, part_count, cmp_def));
 	invalidate_tree_iterator(&it->tree_iterator);
 	it->current.tuple = NULL;
+	if (USE_HINT)
```

Comment:
why not `tree_iterator_set_current_hint` ?

Reply:
This would rely on the fact that `tree_iterator_set_current_hint` doesn't free the old "chunk" if `current_func_key` is NULL. This wouldn't improve readability. This is a constructor so setting fields directly is okay here so I'd prefer to leave it as is - more straightforward to the reader.